subsys/settings/src/settings.c

```diff
@@ -234,39 +242,80 @@ int settings_commit(void)
 	return settings_commit_subtree(NULL);
 }
 
+static bool skip_commit(int handler_cprio, int cprio, int *next_cprio)
+{
+	if (handler_cprio < cprio) {
+		return true;
+	}
+
+	if (handler_cprio == cprio) {
+		return false;
+	}
+
+	if (*next_cprio == cprio) {
+		*next_cprio = handler_cprio;
+		return true;
+	}
```

Comment:
Not sure I understand this. When is this ever true?

Reply:
At the start of the loop the next_cprio is set to cprio. At the end of the loop it checks if the next_prio is still equal to cprio, if this is the case it breaks out of the loop because there are no more handlers with a higher cprio.
The first time a handler is found with higher prio the next prio it is set to that value, subsequent calls can reduce this if the handler has a lower priority value.